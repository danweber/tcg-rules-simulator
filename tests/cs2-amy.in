VERSION:2
# [Start of Your Turn] If you have a monster, gain 1 memory.
# [Main] By suspending this Tamer, until the end of your opponent's turn, 1 of your opponent's Monsters gains \"[Start of Your Main Phase] Attack with this Monster\"."
FIELD2:Amy1,CS2-Aggr Mon,CS2-02
HAND2:CS2-Aggr
# phoenix in sources, shouldn't activate
FIELD1:Tauntee,CS1-10,CS1-08,CS1-03
EGGZONE1:Other,CS1-05,CS1-04
SECURITY2:CS2-02
HAND1:CS1-02 CS1-02
DECK1:CS1-02

GAME:P2 T2 HATCHING 9
DUMP:EGGZONE,EGGS
CMD:MAIN
DUMP:EGGZONE,EGGS

CMD:PLAY CS2-Aggr Amy2
EFFECT-TREE
GET-CHOICES
DUMP

ECHO: -- activate amy we just played
CMD:MAIN Amy2
EFFECT-TREE
GET-CHOICES
DUMP
STEP:TO:0
EFFECT-TREE
GET-CHOICES
DUMP

ECHO: -- activate the other
CMD:MAIN Amy1
EFFECT-TREE
GET-CHOICES
DUMP
STEP:TO:0
EFFECT-TREE
GET-CHOICES
DUMP

CMD:PASS
DUMP

CMD:RAISE
EFFECT-TREE
GET-CHOICES
DUMP:DP,SECURITY

ECHO: -- choose either amy forced effect
STEP:TO:0
EFFECT-TREE
GET-CHOICES
DUMP:DP,SECURITY

EXIT

