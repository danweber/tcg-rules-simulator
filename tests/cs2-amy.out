FIELD1:Tauntee,CS1-10,CS1-08,CS1-03
FIELD2:Amy1,CS2-17 Mon,CS2-02
EGGS1:
EGGS2:
EGGZONE1:Other,CS1-05,CS1-04
EGGZONE2:
GAME:P2 T2 HATCHING 9
FIELD1:Tauntee,CS1-10,CS1-08,CS1-03
FIELD2:Amy1,CS2-17 Mon,CS2-02
EGGS1:
EGGS2:
EGGZONE1:Other,CS1-05,CS1-04
EGGZONE2:
GAME:P2 T2 MAIN 10
EFFECT-TREE:NOTHING
GET-CHOICES:User option:
FIELD1:Tauntee,CS1-10,CS1-08,CS1-03
FIELD2:Amy1,CS2-17 Mon,CS2-02 Amy2,CS2-17
GAME:P2 T2 MAIN 7
ECHO: -- activate amy we just played
EFFECT-TREE:RES DEPTH 1 PROCESSING:Aggressive Amy
GET-CHOICES:Choose:
GET-CHOICES:Activate Aggressive Amy suspending
GET-CHOICES:Don't pay/activate
FIELD1:Tauntee,CS1-10,CS1-08,CS1-03
FIELD2:Amy1,CS2-17 Mon,CS2-02 Amy2,CS2-17
GAME:P2 T2 MAIN 7
EFFECT-TREE:NOTHING
GET-CHOICES:Choose:
FIELD1:Tauntee,CS1-10,CS1-08,CS1-03
FIELD2:Amy1,CS2-17 Mon,CS2-02 Amy2,CS2-17,REST
GAME:P2 T2 MAIN 7
ECHO: -- activate the other
EFFECT-TREE:RES DEPTH 1 PROCESSING:Aggressive Amy
GET-CHOICES:Choose:
GET-CHOICES:Activate Aggressive Amy suspending
GET-CHOICES:Don't pay/activate
FIELD1:Tauntee,CS1-10,CS1-08,CS1-03
FIELD2:Amy1,CS2-17 Mon,CS2-02 Amy2,CS2-17,REST
GAME:P2 T2 MAIN 7
EFFECT-TREE:NOTHING
GET-CHOICES:Choose:
FIELD1:Tauntee,CS1-10,CS1-08,CS1-03
FIELD2:Amy1,CS2-17,REST Mon,CS2-02 Amy2,CS2-17,REST
GAME:P2 T2 MAIN 7
FIELD1:Tauntee,CS1-10,CS1-08,CS1-03
FIELD2:Amy1,CS2-17,REST Mon,CS2-02 Amy2,CS2-17,REST
GAME:P1 T3 HATCHING 3
EFFECT-TREE:RES DEPTH 1 Aggressive Amy, Aggressive Amy
GET-CHOICES:Choose effect to process:
GET-CHOICES:Aggressive Amy [Start of Your Main Phase] Attack with this Monster
GET-CHOICES:Aggressive Amy [Start of Your Main Phase] Attack with this Monster
FIELD1:Tauntee,CS1-10,CS1-08,CS1-03,13000DP Other,CS1-05,CS1-04,6500DP
FIELD2:Amy1,CS2-17,REST,NaNDP Mon,CS2-02,2000DP Amy2,CS2-17,REST,NaNDP
SECURITY1:
SECURITY2:CS2-02
GAME:P1 T3 START_OF_MAIN 3
ECHO: -- choose either amy forced effect
EFFECT-TREE:NOTHING
GET-CHOICES:Choose effect to process:
FIELD1:Tauntee,CS1-10,CS1-08,CS1-03,REST,13000DP Other,CS1-05,CS1-04,6500DP
FIELD2:Amy1,CS2-17,REST,NaNDP Amy2,CS2-17,REST,NaNDP
SECURITY1:
SECURITY2:
GAME:P1 T3 MAIN 3
